doc/specs/#1502 - Advanced Tab Switcher/spec.md

```diff
@@ -0,0 +1,230 @@
+
+In the box will be a vertical list of tabs with their titles and their assigned number for quick switching, and only one line will be highlighted to signify that tab is currently selected. The top 9 tabs in the list are numbered for quick switching, and the rest of the tabs will simply have an empty space where a number would be.
+
+The list would look (roughly) like this:
+```
+1 foo (highlighted)
+2 boo
+3 Windows boofoo
+4 /c/Users/booboo
+5 Git Moo
+6 shoo
+7 /c/
+8 /d/
+9 /e/
+  /f/
+  /g/
+  /h/
+```
+
+The highlighted line can move up or down, and if the user moves up while the highlighted line is already at the top of the list, the highlight will wrap around to the bottom of the list. Similarly, it will wrap to the top if the highlight is at the bottom of the list and the user moves down.
+
+If there's more tabs than the UI can display, the list of tabs will scroll up/down as the user keeps iterating up/down. Even if some of the numbered tabs (the first 9 tabs) are not visible, the user can still press any number 1 through 9 to quick switch to that tab.
+
+To give an example of what happens after scrolling past the end, imagine a user is starting from the state in the mock above. The user then iterates down past the end of the visible list four times. The below mock shows the result.
+
+```
+5 Git Moo
+6 shoo
+7 /c/
+8 /d
+9 /e/
+  /f/
+  /g/
+  /h/
+  /i/
+  /j/
+  /k/
+  /l/ (highlighted)
+```
+
+The tabs designated by numbers 1 through 4 are no longer visible (but still quick-switchable), and the list now starts with "Git Moo", which is associated with number 5.
+
+### Using the Switcher
+
+#### Opening the Tab Switcher
+
+The user will press a keybinding named `openTabSwitcher` to bring up the UI.
+The user will be able to change it to whatever they like.
+There will also be an optional `anchor` arg that may be provided to this keybinding. 
+
+#### Keeping it open
+
+We use the term `anchor` to illustrate the idea that the UI stays visible as long as something is "anchoring" it down.
+
+Here's an example of how to set the `anchor` key in the settings:
+```
+  {"keys": ["ctrl+tab"], "command": {"action": "openTabSwitcher", "anchor": "ctrl" }}
+```
+
+This user provided the `anchor` key arg, and set it to <kbd>ctrl</kbd>. So, the user would open the UI with <kbd>ctrl+tab</kbd>, and as long as the user is holding <kbd>ctrl</kbd> down, the UI won't dismiss. The moment the user releases <kbd>ctrl</kbd>, the UI dismisses. The `anchor` key needs to be one of the keys in the `openTabSwitcher` keybinding. If it isn't, we'll display a warning dialog in this case saying that the `anchor` key isn't actually part of the keybinding, and the user might run into some weird behavior.
+
+If `openTabSwitcher` is not given an `anchor` key, the switcher will stay visible even after the release of the keybinding.
+
+#### Switching through Tabs

```

Comment:
Yup! The pointer interactions would be handled the same way, also since I'll be using the command palette UI for this, whatever interactions are available for the command palette will be available when using the tab switcher. I'll just add some "custom" interactivity when the palette is in "tab switcher mode". 
Hovering over an entry will make it highlighted, and the tab preview idea is in my future considerations section! 😄